apdbPlots.py

```diff
@@ -0,0 +1,599 @@
+"""
+
+
+def addVisitCcdToSrcTable(sourceTable):
+    """Add visit and ccd columns to sourceTable dataframe.
+
+    This assumes the `ccdVisitId` field may be cast to type `str` and
+    is arranged such that the last 2 characters represent a ccd and all
+    preceding characters represent a visit.
+
+    TODO: generalize this for other exposure ID formats.
+
+    Parameters
+    ----------
+    sourceTable : `pandas.core.frame.DataFrame`
+        Pandas dataframe with DIA Sources from an APDB.
+
+    Returns
+    -------
+    sourceTable : `pandas.core.frame.DataFrame`
+        The same as the input sourceTable, with new visit and ccd columns.
+    """
+    sourceTable['ccd'] = sourceTable.ccdVisitId.apply(lambda x: str(x)[-2:])
+    sourceTable['visit'] = sourceTable.ccdVisitId.apply(lambda x: str(x)[:-2])
+    return sourceTable
+
+
+def plot2axes(x1, y1, x2, y2, xlabel, ylabel1, ylabel2, y1err=None, y2err=None, title=''):
+    """Generic plot framework for showing one y-variable on the left
+    and another on the right. The x-axis is shared on the bottom.
+
+    Parameters
+    ----------
+    x1 : `list` or `array`
+    y1 : `list` or `array`
+    x2 : `list` or `array`
+    y2 : `list` or `array`
+    xlabel : `str`
+        Label for shared x axis
+    ylabel1 : `str`
+        Label for y1 axis
+    ylabel2 : `str`
+        Label for y2 axis
+    title : `str`
+        Title for the plot, optional.
+    """
+    fig, ax1 = plt.subplots(figsize=(9, 3))
+    color = 'C1'
+    if y1err is not None:
+        ax1.errorbar(x1, y1, yerr=y1err, color=color, marker='o', ls=':')
+    else:
+        ax1.plot(x1, y1, color=color, marker='o', ls=':')
+    ax1.set_xlabel(xlabel)
+    ax1.set_ylabel(ylabel1, color=color)
+    ax1.tick_params(axis='y', labelcolor=color)
+    for label in ax1.get_xticklabels():
+        label.set_ha("right")
+        label.set_rotation(45)
+        label.set_size("smaller")
+    ax2 = ax1.twinx()  # instantiate a second axes that shares the same x-axis
+    color = 'C0'
+    ax2.set_ylabel(ylabel2, color=color)  # we already handled the x-label above
+    if y2err is not None:
+        ax2.errorbar(x2, y2, yerr=y2err, color=color, marker='o', ls=':')
+    else:
+        ax2.plot(x2, y2, color=color, marker='o', ls=':')
+    ax2.tick_params(axis='y', labelcolor=color)
+    plt.title(title)
+    fig.tight_layout()  # otherwise the right y-label is slightly clipped
+
+
+def plotDiaSourcesPerVisit(sourceTable, approxVisitArea=0.045, title=''):
+    """Plot DIA Sources per visit.
+
+    The plot will have two y-axes: number of DIA Sources per square degree and
+    median FWHM per ixx or iyy in pixels.
+
+    Parameters
+    ----------
+    sourceTable : `pandas.core.frame.DataFrame`
+        Pandas dataframe with DIA Sources from an APDB.
+    approxVisitArea : `float`
+        Approximate area of one visit (all ccds), in square degrees.
+        Default set to an appropriate DECam value.
```

Comment:
It would be useful to add the default value in numbers as well when you mention it in text.